### ISSUE TITLE
Popup not triggering on front-page.php template

### ISSUE THREAD

**Issue by CostaC** [elementor/elementor#15579]
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor. 

## Description
I'm using a custom front-page.php that doesn't use Elementor for page contents.

Via Elementor Display Conditions, I've assigned a popup to "Front Page" and to "About" page.  The popup triggers on the "About" page but not on the front page.

I've used Query Monitor to confirm that "is_front_page() == true" on the home page.

If I set the Display Conditions to "Entire Site" the popup *will* display on the front page.  But it's not triggering when I specifically set   Display Condition to "Front Page".

I also made sure no other popups were set to trigger on Front Page.

## Steps to reproduce

- Use a custom front-page.php file in theme root
- Set WP > Settings > Reading > Your homepage displays > Latest posts 
- Create Elementor Popup > Display Conditions > Front Page

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: Percona Server (GPL), Release '37', Revision '7c516e9' v5.7.34-37
	PHP Version: 7.4.18
	PHP Max Input Vars: 10000
	PHP Max Post Size: 200M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.7.2
	Site URL: --------
	Home URL: --------
	WP Multisite: No
	Max Upload Size: 200 MB
	Memory limit: 512M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: America/Anchorage
	Debug Mode: Inactive

== Theme ==
	Name: CORE
	Version: 2.0.1
	Author: Core Media
	Child Theme: Yes
	Parent Theme Name: Genesis
	Parent Theme Version: 3.3.3
	Parent Theme Author: StudioPress

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36

== Active Plugins ==
	Announcer
		Version: 5.1
		Author: Aakash Chakravarthy

	Black Studio TinyMCE Widget
		Version: 2.6.9
		Author: Black Studio

	Classic Editor
		Version: 1.6
		Author: WordPress Contributors

	CTC Courses
		Version: 1.1
		Author: David Mattice

	Duplicate Page
		Version: 4.4
		Author: mndpsingh287

	Elementor
		Version: 3.2.5
		Author: Elementor.com

	Elementor Pro
		Version: 3.3.1
		Author: Elementor.com

	Genesis Responsive Slider
		Version: 1.0.1
		Author: StudioPress

	Genesis Simple Edits
		Version: 2.3.1
		Author: StudioPress

	Granular Controls Elementor
		Version: 1.0.5
		Author: Zulfikar Nore

	Gravity Forms
		Version: 2.5.6
		Author: Gravity Forms

	Imagify
		Version: 1.9.14
		Author: WP Media

	ManageWP - Worker
		Version: 4.9.8
		Author: GoDaddy

	MCE Table Buttons
		Version: 3.3
		Author: Jake Goldman, 10up, Oomph

	Redirection
		Version: 5.1.1
		Author: John Godley

	Regenerate Thumbnails
		Version: 3.1.5
		Author: Alex Mills (Viper007Bond)

	Responsive Lightbox & Gallery
		Version: 2.3.2
		Author: dFactory

	Simple Social Icons
		Version: 3.0.2
		Author: StudioPress

	WP Engine Smart Plugin Manager
		Version: 4.20.0
		Author: WP Engine

	WP Mailto Links - Hide & Protect Emails
		Version: 3.1.2
		Author: Ironikus

	WP Rocket
		Version: 3.9.0.5
		Author: WP Media

	Yoast SEO
		Version: 16.6.1
		Author: Team Yoast


== Must-Use Plugins ==
	Force Strong Passwords - WPE Edition
		Version: 1.6.4
		Author: Jason Cosper

	ManageWP - Worker Loader
		Version: 
		Author: ManageWP

	WPE ElasticPress Autosuggest Logger
		Version: 1.0.0
		Author: WP Engine

	WP Engine Seamless Login Plugin
		Version: 1.5.5
		Author: WP Engine

	WP Engine Security Auditor
		Version: 1.0.10
		Author: wpengine

	WP Engine System
		Version: 4.1.0
		Author: WP Engine


== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible
	Granular Controls Elementor: Compatibility not specified

== Elementor Pro - Compatibility Tag ==
	


== Elements Usage ==
	
	page : 6
		accordion : 15
		button : 5
		divider : 13
		heading : 40
		html : 1
		icon-box : 18
		image : 6
		spacer : 3
		text-editor : 21
	popup : 10
		button : 6
		heading : 25
		spacer : 6
	section : 5
		button : 4
		divider : 3
		heading : 11
		icon-box : 3
		text-editor : 19
	wp-page : 9
		button : 5
		divider : 17
		heading : 72
		html : 3
		icon-box : 18
		image : 7
		text-editor : 47


== Log ==
	
Log: showing 20 of 412021-02-22 03:11:28 [info] Elementor Pro/Upgrades - _v_3_1_0_testimonial_carousel Finished 
2021-02-22 03:11:28 [info] Elementor Pro/Upgrades - _v_3_1_0_slides Start  
2021-02-22 03:11:28 [info] Elementor Pro/Upgrades - _v_3_1_0_slides Finished 
2021-02-22 03:11:28 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.0.10',
  'to' => '3.1.0',
)]
2021-03-15 04:22:00 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.1.1',
  'to' => '3.1.4',
)]
2021-04-26 11:07:56 [info] elementor::elementor_updater Started 
2021-04-26 11:07:56 [info] Elementor/Upgrades - _v_3_2_0_migrate_breakpoints_to_new_system Start  
2021-04-26 11:07:56 [info] Elementor/Upgrades - _v_3_2_0_migrate_breakpoints_to_new_system Finished 
2021-04-26 11:07:56 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.1.4',
  'to' => '3.2.2',
)]
2021-04-26 11:07:56 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.1.4',
  'to' => '3.2.2',
)]
2021-05-09 22:12:23 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.2.2',
  'to' => '3.2.3',
)]
2021-05-30 22:14:35 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.2.3',
  'to' => '3.2.4',
)]
2021-06-20 22:30:16 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.2.4',
  'to' => '3.2.5',
)]
2021-06-22 09:11:12 [info] elementor-pro::elementor_pro_updater Started 
2021-06-22 09:11:12 [info] Elementor Pro/Upgrades - _v_3_3_0_nav_menu_icon Start  
2021-06-22 09:11:12 [info] Elementor Pro/Upgrades - _v_3_3_0_nav_menu_icon Finished 
2021-06-22 09:11:12 [info] Elementor Pro/Upgrades - _v_3_3_0_recalc_usage_data Start  
2021-06-22 09:11:12 [info] Elementor Pro/Upgrades - _v_3_3_0_recalc_usage_data Finished 
2021-06-22 09:11:12 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.1.0',
  'to' => '3.3.1',
)]
2021-06-22 09:11:12 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.1.0',
  'to' => '3.3.1',
)]

PHP: showing 1 of 1PHP: 2021-02-03 17:41:29 [notice X 5][/wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]


				```
</details>

**Comment by Aviad-Herman** [comment]
Hi seems like a bug, thanks for reporting this!

**Comment by Aviad-Herman** [comment]
Hi,
We researched this issue and it turns out it is by design.
Only a small number of people today use homepage = latest posts and fixing this issue will likely cause many other unpredictable bugs. 

I can provide these alternatives:
1. Make the popup target Archives --> Posts Archive (I tested this a little but and it seems to work but you will have to test it doesn't make the popup appear in unwanted places)
2. Create a static page that contains your latest posts (using the posts widget) and make the popup target that static page.

Hope this helps.
